apps/web-client/src/routes/orders/suppliers/reconcile/[id]/+page.ts

```diff
@@ -0,0 +1,47 @@
+import { getReconciliationOrder, getReconciliationOrderLines } from "$lib/db/cr-sqlite/order-reconciliation";
+import type { PageLoad } from "./$types";
+import { getPlacedSupplierOrderLinesForReconciliation } from "$lib/db/cr-sqlite/suppliers";
+import type { BookEntry } from "@librocco/db";
+
+export const load: PageLoad = async ({ parent, params, depends }) => {
+	depends("reconciliationOrder:data");
+
+	const { ordersDb } = await parent();
+	const reconciliationOrder = await getReconciliationOrder(ordersDb, parseInt(params.id));
+	const reconciliationOrderLines = await getReconciliationOrderLines(ordersDb, parseInt(params.id));
+
+	const supplierOrderIds = JSON.parse(reconciliationOrder.supplier_order_ids) || [];
```

Comment:
`getReconciliationOrder` should parse this JSON and throw if its invalid. The `[]` default should not be returned

Reply:
there is a check in the db for invalid json. will remove the [] though

Reply:
The call to `JSON.parse` doesn't know what has happened elsewhere. It should still be handled properly. It makes the most sense for it to happen in the handler, where other cases are being handled